FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_IP.c

```diff
@@ -2210,7 +2221,7 @@ size_t uxDataLengthBytes = uxByteCount;
 	}
 
 	/* If half-word (16-bit) aligned... */
-	if( ( ( ulAlignBits == 1U ) || ( ulAlignBits == 2U ) ) && ( uxDataLengthBytes >= 2U ) )
+	if( ( ( uxAlignBits == 1U ) || ( uxAlignBits == 2U ) ) && ( uxDataLengthBytes >= 2U ) )
```

Comment:
If it is aligned to a 16-bit boundary, wouldn't the test be?

```c
( uxAlignBits == 2UL ) || ( uxAlignBits == 0UL )
```

Reply:
My understanding of this code is that the preceding code ensures that byte aligned buffers are half-word aligned and then this part of the code handles half-word aligned buffers that are originally that way or have become that way because of the preceding code.

The data length check is just a guard to ensure that there's at least a half word of data.

Buffers that are already word aligned (uxAlignBits == 0UL) don't need any special treatment before the parallel summing step.